src/main/java/org/jabref/model/util/FileHelper.java

```diff
@@ -0,0 +1,123 @@
+package org.jabref.model.util;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.metadata.FileDirectoryPreferences;
+
+public class FileHelper {
+    private static final Pattern SLASH = Pattern.compile("/");
+    private static final Pattern BACKSLASH = Pattern.compile("\\\\");
+
+    /**
+     * Returns the extension of a file or Optional.empty() if the file does not have one (no . in name).
+     *
+     * @param file
+     * @return The extension, trimmed and in lowercase.
+     */
+    public static Optional<String> getFileExtension(File file) {
+        return getFileExtension(file.getName());
+    }
+
+    /**
+     * Returns the extension of a file name or Optional.empty() if the file does not have one (no "." in name).
+     *
+     * @param fileName
+     * @return The extension (without leading dot), trimmed and in lowercase.
+     */
+    public static Optional<String> getFileExtension(String fileName) {
```

Comment:
Can you add a JavaDoc comment why you implemented that method on your own instead of using [Files.getFileExtension(fileName);](http://stackoverflow.com/a/15985486/873282)? We have the Guava libraries and should use them 😇